### PR TITLE
Make different Markdown files have different titles in Doxygen output

### PR DESCRIPTION
**Pull request by jafingerhut**

### CODE REVIEW

**Comment by jafingerhut** [comment]
@fruffy @AdarshRawat1 The current Doxygen-generated page here has about 4 or 5 "README" headings, and a few others that are similarly non-obvious what they are intended to be. These changes should make them all have more descriptive top level headings.

There is no difference between this PR and the other one I closed recently, except this one has the correct DCO text included in the commit.

**Comment by AdarshRawat1** [comment]
@jafingerhut Thank you for tagging the pages, this will improve Documentation .

**Review by fruffy — APPROVED** [review]

File: backends/p4tools/modules/testgen/lib/README.md [hunk 1]
```diff
@@ -1 +1,3 @@
+# P4testgen Library
```
Comment (fruffy) [inline backends/p4tools/modules/testgen/lib/README.md:1]:
```suggestion
# P4Testgen Library
```

**Review by fruffy — CHANGES_REQUESTED** [review]

File: backends/p4tools/modules/testgen/lib/README.md (hunk 1 above)
Comment (jafingerhut) [reply backends/p4tools/modules/testgen/lib/README.md]: done

**Review by fruffy — APPROVED** [review]

**Pull request merged**